File: YonoSBI_Flutter/lib/screens/account_summary_screen.dart
import 'package:flutter/material.dart';
import '../constants/theme.dart';

class AccountSummaryScreen extends StatelessWidget {
  const AccountSummaryScreen({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: AppColors.background,
      body: SafeArea(
        child: Column(
          children: [
            // Header
            Container(
              padding: const EdgeInsets.all(AppSpacing.base),
              decoration: const BoxDecoration(
                gradient: LinearGradient(
                  colors: [AppColors.primary, AppColors.primaryDark],
                ),
              ),
              child: Row(
                children: [
                  IconButton(
                    icon: const Icon(Icons.arrow_back, color: AppColors.white),
                    onPressed: () => Navigator.pop(context),
                  ),
                  const Text(
                    'Accounts',
                    style: TextStyle(
                      fontSize: AppFontSizes.lg,
                      fontWeight: FontWeight.w600,
                      color: AppColors.white,
                    ),
                  ),
                ],
              ),
            ),
            
            Expanded(
              child: SingleChildScrollView(
                child: Column(
                  children: [
                    // Account Card
                    Container(
                      margin: const EdgeInsets.all(AppSpacing.base),
                      padding: const EdgeInsets.all(AppSpacing.lg),
                      decoration: BoxDecoration(
                        gradient: const LinearGradient(
                          colors: [Color(0xFF6B3FA0), Color(0xFF9B6BC3), Color(0xFF7B4AAB)],
                        ),
                        borderRadius: BorderRadius.circular(AppBorderRadius.xl),
                      ),
                      child: Column(
                        crossAxisAlignment: CrossAxisAlignment.start,
                        children: [
                          const Text(
                            'SAVINGS A/C',
                            style: TextStyle(
                              fontSize: AppFontSizes.xs,
                              color: AppColors.white,
                              letterSpacing: 0.5,
                            ),
                          ),
                          const SizedBox(height: AppSpacing.md),
                          Row(
                            children: [
                              const Text(
                                'ACCOUNT NUMBER',
                                style: TextStyle(
                                  fontSize: AppFontSizes.xs,
                                  color: AppColors.white,
                                ),
                              ),
                              const SizedBox(width: 8),
                              const Icon(Icons.visibility_off, size: 16, color: AppColors.white),
                            ],
                          ),
                          const SizedBox(height: 4),
                          const Text(
                            '44723514860',
                            style: TextStyle(
                              fontSize: AppFontSizes.lg,
                              fontWeight: FontWeight.w600,
                              color: AppColors.white,
                            ),
                          ),
                          const SizedBox(height: AppSpacing.md),
                          const Text(
                            'AVAILABLE BALANCE',
                            style: TextStyle(
                              fontSize: AppFontSizes.xs,
                              color: AppColors.white,
                            ),
                          ),
                          const SizedBox(height: 4),
                          const Text(
                            '₹75.10',
                            style: TextStyle(
                              fontSize: AppFontSizes.xxxl,
                              fontWeight: FontWeight.bold,
                              color: AppColors.white,
                            ),
                          ),
                          const SizedBox(height: AppSpacing.lg),
                          Row(
                            children: [
                              Expanded(
                                child: ElevatedButton(
                                  onPressed: () {},
                                  style: ElevatedButton.styleFrom(
                                    backgroundColor: Colors.white.withOpacity(0.2),
                                    foregroundColor: AppColors.white,
                                    elevation: 0,
                                  ),
                                  child: const Text('Manage Account'),
                                ),
                              ),
                              const SizedBox(width: 12),
                              Expanded(
                                child: OutlinedButton(
                                  onPressed: () {},
                                  style: OutlinedButton.styleFrom(
                                    foregroundColor: AppColors.white,
                                    side: const BorderSide(color: AppColors.white),
                                  ),
                                  child: const Text('View Debit Card'),
                                ),
                              ),
                            ],
                          ),
                        ],
                      ),
                    ),
                    
                    // Summary Section
                    Container(
                      margin: const EdgeInsets.symmetric(horizontal: AppSpacing.base),
                      decoration: BoxDecoration(
                        color: AppColors.white,
                        borderRadius: BorderRadius.circular(AppBorderRadius.lg),
                      ),
                      child: Column(
                        children: [
                          Container(
                            padding: const EdgeInsets.all(AppSpacing.base),
                            decoration: const BoxDecoration(
                              border: Border(
                                bottom: BorderSide(color: AppColors.borderLight),
                              ),
                            ),
                            child: const Row(
                              children: [
                                Text(
                                  'Summary',
                                  style: TextStyle(
                                    fontSize: AppFontSizes.md,
                                    fontWeight: FontWeight.w600,
                                    color: AppColors.textPrimary,
                                  ),
                                ),
                              ],
                            ),
                          ),
                          _buildSummaryRow('Available Balance', '₹75.10'),
                          _buildSummaryRow('Hold/Lien Amount', '₹0.00'),
                          _buildSummaryRow('Uncleared Balance', '₹0.00'),
                          _buildSummaryRow('MOD Balance', '₹0.00'),
                        ],
                      ),
                    ),
                    
                    const SizedBox(height: AppSpacing.base),
                    
                    // Account Details
                    Container(
                      margin: const EdgeInsets.symmetric(horizontal: AppSpacing.base),
                      decoration: BoxDecoration(
                        color: AppColors.white,
                        borderRadius: BorderRadius.circular(AppBorderRadius.lg),
                      ),
                      child: Column(
                        children: [
                          _buildDetailRow('Account Description', 'LOTUS SAVING BANK-ADHAR-CHQ'),
                          _buildDetailRow('Mode of Operation', 'Single'),
                          _buildDetailRow('Currency', 'Rupees'),
                          _buildDetailRow('Rate of Interest', '2.5%'),
                          _buildDetailRowWithLink('Nominee(s)', 'View Details'),
                        ],
                      ),
                    ),
                    
                    const SizedBox(height: AppSpacing.xxl),
                  ],
                ),
              ),
            ),
          ],
        ),
      ),
    );
  }

  Widget _buildSummaryRow(String label, String value) {
    return Container(
      padding: const EdgeInsets.all(AppSpacing.base),
      decoration: const BoxDecoration(
        border: Border(
          bottom: BorderSide(color: AppColors.borderLight),
        ),
      ),
      child: Row(
        mainAxisAlignment: MainAxisAlignment.spaceBetween,
        children: [
          Text(
            label,
            style: const TextStyle(
              fontSize: AppFontSizes.sm,
              color: AppColors.textSecondary,
            ),
          ),
          Text(
            value,
            style: const TextStyle(
              fontSize: AppFontSizes.sm,
              fontWeight: FontWeight.w500,
              color: AppColors.textPrimary,
            ),
          ),
        ],
      ),
    );
  }

  Widget _buildDetailRow(String label, String value) {
    return Container(
      padding: const EdgeInsets.all(AppSpacing.base),
      decoration: const BoxDecoration(
        border: Border(
          bottom: BorderSide(color: AppColors.borderLight),
        ),
      ),
      child: Row(
        mainAxisAlignment: MainAxisAlignment.spaceBetween,
        children: [
          Text(
            label,
            style: const TextStyle(
              fontSize: AppFontSizes.sm,
              color: AppColors.textSecondary,
            ),
          ),
          Expanded(
            child: Text(
              value,
              textAlign: TextAlign.right,
              style: const TextStyle(
                fontSize: AppFontSizes.sm,
                fontWeight: FontWeight.w500,
                color: AppColors.textPrimary,
              ),
            ),
          ),
        ],
      ),
    );
  }

  Widget _buildDetailRowWithLink(String label, String linkText) {
    return Container(
      padding: const EdgeInsets.all(AppSpacing.base),
      decoration: const BoxDecoration(
        border: Border(
          bottom: BorderSide(color: AppColors.borderLight),
        ),
      ),
      child: Row(
        mainAxisAlignment: MainAxisAlignment.spaceBetween,
        children: [
          Text(
            label,
            style: const TextStyle(
              fontSize: AppFontSizes.sm,
              color: AppColors.textSecondary,
            ),
          ),
          Text(
            linkText,
            style: const TextStyle(
              fontSize: AppFontSizes.sm,
              fontWeight: FontWeight.w500,
              color: AppColors.primary,
            ),
          ),
        ],
      ),
    );
  }
}

File: src/screens/AccountSummaryScreen.js
import React from 'react';
import {
    View,
    Text,
    StyleSheet,
    TouchableOpacity,
    ScrollView,
    StatusBar,
} from 'react-native';
import { LinearGradient } from 'expo-linear-gradient';
import { Ionicons } from '@expo/vector-icons';
import { Colors, FontSizes, FontWeights, Spacing, BorderRadius, Shadows } from '../constants/theme';

const AccountSummaryScreen = ({ navigation }) => {
    const accountData = {
        type: 'SAVINGS A/C',
        number: '44723514860',
        balance: 75.10,
        availableBalance: 75.10,
        holdAmount: 0.00,
        unclearedBalance: 0.00,
        modBalance: 0.00,
        description: 'LOTUS SAVING BANK-ADHAR-CHQ',
        modeOfOperation: 'Single',
        currency: 'Rupees',
        rateOfInterest: '2.5%',
    };

    return (
        <View style={styles.container}>
            <StatusBar barStyle="light-content" backgroundColor={Colors.primary} />

            {/* Header */}
            <LinearGradient
                colors={[Colors.primary, Colors.primaryDark]}
                style={styles.header}
            >
                <TouchableOpacity style={styles.backBtn} onPress={() => navigation.goBack()}>
                    <Ionicons name="chevron-back" size={24} color={Colors.textWhite} />
                </TouchableOpacity>
                <Text style={styles.headerTitle}>Accounts</Text>
                <View style={{ width: 32 }} />
            </LinearGradient>

            <ScrollView style={styles.content}>
                {/* Account Card */}
                <LinearGradient
                    colors={['#6B3FA0', '#9B6BC3', '#7B4AAB']}
                    start={{ x: 0, y: 0 }}
                    end={{ x: 1, y: 1 }}
                    style={styles.accountCard}
                >
                    <Text style={styles.accountType}>{accountData.type}</Text>
                    <View style={styles.accountNumberRow}>
                        <Text style={styles.accountNumberLabel}>ACCOUNT NUMBER</Text>
                        <Ionicons name="eye-off-outline" size={16} color={Colors.textWhite} />
                    </View>
                    <Text style={styles.accountNumber}>{accountData.number}</Text>

                    <Text style={styles.balanceLabel}>AVAILABLE BALANCE</Text>
                    <Text style={styles.balanceAmount}>₹{accountData.balance.toFixed(2)}</Text>

                    <View style={styles.cardActions}>
                        <TouchableOpacity style={styles.cardActionBtn}>
                            <Text style={styles.cardActionText}>Manage Account</Text>
                        </TouchableOpacity>
                        <TouchableOpacity style={[styles.cardActionBtn, styles.cardActionBtnOutline]}>
                            <Text style={styles.cardActionTextOutline}>View Debit Card</Text>
                        </TouchableOpacity>
                    </View>
                </LinearGradient>

                {/* Summary Section */}
                <View style={styles.summarySection}>
                    <View style={styles.summaryHeader}>
                        <Text style={styles.summaryTitle}>Summary</Text>
                    </View>

                    <View style={styles.summaryTable}>
                        <View style={styles.summaryRow}>
                            <Text style={styles.summaryLabel}>Available Balance</Text>
                            <Text style={styles.summaryValue}>₹{accountData.availableBalance.toFixed(2)}</Text>
                        </View>
                        <View style={styles.summaryRow}>
                            <Text style={styles.summaryLabel}>Hold/Lien Amount</Text>
                            <Text style={styles.summaryValue}>₹{accountData.holdAmount.toFixed(2)}</Text>
                        </View>
                        <View style={styles.summaryRow}>
                            <Text style={styles.summaryLabel}>Uncleared Balance</Text>
                            <Text style={styles.summaryValue}>₹{accountData.unclearedBalance.toFixed(2)}</Text>
                        </View>
                        <View style={styles.summaryRow}>
                            <Text style={styles.summaryLabel}>MOD Balance</Text>
                            <Text style={styles.summaryValue}>₹{accountData.modBalance.toFixed(2)}</Text>
                        </View>
                    </View>
                </View>

                {/* Account Details Section */}
                <View style={styles.detailsSection}>
                    <View style={styles.detailRow}>
                        <Text style={styles.detailLabel}>Account Description</Text>
                        <Text style={styles.detailValue}>{accountData.description}</Text>
                    </View>
                    <View style={styles.detailRow}>
                        <Text style={styles.detailLabel}>Mode of Operation</Text>
                        <Text style={styles.detailValue}>{accountData.modeOfOperation}</Text>
                    </View>
                    <View style={styles.detailRow}>
                        <Text style={styles.detailLabel}>Currency</Text>
                        <Text style={styles.detailValue}>{accountData.currency}</Text>
                    </View>
                    <View style={styles.detailRow}>
                        <Text style={styles.detailLabel}>Rate of Interest</Text>
                        <Text style={styles.detailValue}>{accountData.rateOfInterest}</Text>
                    </View>
                    <View style={styles.detailRow}>
                        <Text style={styles.detailLabel}>Nominee(s)</Text>
                        <TouchableOpacity>
                            <Text style={styles.viewDetailsLink}>View Details</Text>
                        </TouchableOpacity>
                    </View>
                </View>
            </ScrollView>
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: Colors.background,
    },
    header: {
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'space-between',
        paddingTop: 44,
        paddingHorizontal: Spacing.lg,
        paddingBottom: Spacing.base,
    },
    backBtn: {
        width: 32,
        height: 32,
        alignItems: 'center',
        justifyContent: 'center',
    },
    headerTitle: {
        fontSize: FontSizes.lg,
        fontWeight: FontWeights.semiBold,
        color: Colors.textWhite,
    },
    content: {
        flex: 1,
    },
    accountCard: {
        margin: Spacing.base,
        borderRadius: BorderRadius.xl,
        padding: Spacing.lg,
        ...Shadows.purple,
    },
    accountType: {
        fontSize: FontSizes.xs,
        color: Colors.textWhite,
        opacity: 0.85,
        letterSpacing: 0.5,
        marginBottom: Spacing.md,
    },
    accountNumberRow: {
        flexDirection: 'row',
        alignItems: 'center',
        gap: 8,
        marginBottom: 4,
    },
    accountNumberLabel: {
        fontSize: FontSizes.xs,
        color: Colors.textWhite,
        opacity: 0.7,
    },
    accountNumber: {
        fontSize: FontSizes.lg,
        fontWeight: FontWeights.semiBold,
        color: Colors.textWhite,
        marginBottom: Spacing.md,
    },
    balanceLabel: {
        fontSize: FontSizes.xs,
        color: Colors.textWhite,
        opacity: 0.7,
        marginBottom: 4,
    },
    balanceAmount: {
        fontSize: FontSizes.xxxl,
        fontWeight: FontWeights.bold,
        color: Colors.textWhite,
    },
    cardActions: {
        flexDirection: 'row',
        gap: 12,
        marginTop: Spacing.lg,
    },
    cardActionBtn: {
        flex: 1,
        paddingVertical: Spacing.sm,
        backgroundColor: 'rgba(255,255,255,0.2)',
        borderRadius: BorderRadius.md,
        alignItems: 'center',
    },
    cardActionBtnOutline: {
        backgroundColor: 'transparent',
        borderWidth: 1,
        borderColor: 'rgba(255,255,255,0.5)',
    },
    cardActionText: {
        fontSize: FontSizes.sm,
        fontWeight: FontWeights.medium,
        color: Colors.textWhite,
    },
    cardActionTextOutline: {
        fontSize: FontSizes.sm,
        fontWeight: FontWeights.medium,
        color: Colors.textWhite,
    },
    summarySection: {
        backgroundColor: Colors.white,
        marginHorizontal: Spacing.base,
        borderRadius: BorderRadius.lg,
        overflow: 'hidden',
    },
    summaryHeader: {
        paddingVertical: Spacing.md,
        paddingHorizontal: Spacing.base,
        borderBottomWidth: 1,
        borderBottomColor: Colors.borderLight,
    },
    summaryTitle: {
        fontSize: FontSizes.md,
        fontWeight: FontWeights.semiBold,
        color: Colors.textPrimary,
    },
    summaryTable: {},
    summaryRow: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        paddingVertical: Spacing.md,
        paddingHorizontal: Spacing.base,
        borderBottomWidth: 1,
        borderBottomColor: Colors.borderLight,
    },
    summaryLabel: {
        fontSize: FontSizes.sm,
        color: Colors.textSecondary,
    },
    summaryValue: {
        fontSize: FontSizes.sm,
        fontWeight: FontWeights.medium,
        color: Colors.textPrimary,
    },
    detailsSection: {
        backgroundColor: Colors.white,
        marginHorizontal: Spacing.base,
        marginTop: Spacing.base,
        marginBottom: Spacing.xxl,
        borderRadius: BorderRadius.lg,
        overflow: 'hidden',
    },
    detailRow: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
        paddingVertical: Spacing.md,
        paddingHorizontal: Spacing.base,
        borderBottomWidth: 1,
        borderBottomColor: Colors.borderLight,
    },
    detailLabel: {
        fontSize: FontSizes.sm,
        color: Colors.textSecondary,
    },
    detailValue: {
        fontSize: FontSizes.sm,
        fontWeight: FontWeights.medium,
        color: Colors.textPrimary,
        textAlign: 'right',
        flex: 1,
        marginLeft: Spacing.base,
    },
    viewDetailsLink: {
        fontSize: FontSizes.sm,
        fontWeight: FontWeights.medium,
        color: Colors.primary,
    },
});

export default AccountSummaryScreen;
